lib/spack/spack/compilers/msvc.py

```diff
@@ -29,6 +29,78 @@
 }
 
 
+class CmdCall:
+    def __init__(self, *cmds):
+        self._cmds = cmds
+
+    def __call__(self):
+        if not self._cmds:
+            raise RuntimeError(
+                """Attempting to run commands from CMD without specifying commands.
+                Please add commands to be run."""
+            )
```

Comment:
I only see this constructed/called in one place, since in that spot all the commands are provided up front I recommend:

* Raising the exception in `__init__`
* Remove the `add_command` function